Quadriga/src/main/java/edu/asu/spring/quadriga/service/textfile/impl/TextFileManager.java

```diff
@@ -0,0 +1,77 @@
+package edu.asu.spring.quadriga.service.textfile.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.spring.quadriga.dao.textfile.ITextFileDAO;
+import edu.asu.spring.quadriga.domain.workspace.ITextFile;
+import edu.asu.spring.quadriga.dto.TextFileDTO;
+import edu.asu.spring.quadriga.exceptions.FileStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.textfile.IFileSaveService;
+import edu.asu.spring.quadriga.service.textfile.ITextFileManager;
+import edu.asu.spring.quadriga.service.textfile.mapper.ITextFileMapper;
+import edu.asu.spring.quadriga.utilities.IFileSaveUtility;
+
+/**
+ * @author Nischal Samji
+ *
+ */
+@Service
+@Transactional
+public class TextFileManager implements ITextFileManager {
+
+    @Autowired
+    private ITextFileDAO txtFileDAO;
+
+    @Autowired
+    private IFileSaveService fileSaveServ;
+
+    @Qualifier("txtfileSaveUtil")
+    @Autowired
+    private IFileSaveUtility fileManager;
+
+    @Autowired
+    private ITextFileMapper tfSMapper;
+
+    
+    @Override
+    public boolean saveTextFile(ITextFile txtFile) throws FileStorageException, QuadrigaStorageException {
+        String txtId = txtFileDAO.generateUniqueID();
+        txtFile.setTextId(txtId);
+        TextFileDTO txtFileDTO = tfSMapper.getTextFileDTO(txtFile);
+        txtFileDTO.setTextId(txtId);
+        txtFileDAO.saveNewDTO(txtFileDTO);
+        return fileSaveServ.saveFileToLocal(txtFile);
+    }
+
+    @Override
+    public List<ITextFile> retrieveTextFiles(String wsId) throws QuadrigaStorageException {
+        List<TextFileDTO> tfDTOList = txtFileDAO.getTextFileDTObyWsId(wsId);
+        List<ITextFile> tfList = new ArrayList<>();
+        for(TextFileDTO tfDTO : tfDTOList)
+        {
+            tfList.add(tfSMapper.getTextFile(tfDTO));
+        }
+        return tfList;
+
+    }
+
+    @Override
+    public String retrieveTextFileContent(String txtId) throws FileStorageException {
+
+        TextFileDTO tfDTO = txtFileDAO.getDTO(txtId);
+        String fileName= tfDTO.getFilename();
+        if(!fileName.contains("."))
+            fileName+=".txt";
+        return fileManager.readFileContent(fileName, tfDTO.getTextId());
```

Comment:
I would go through IFileSaveService again. Give it a textfile and let the IFileSaveService add the text to the object (in another string property) I would also add a file object to the textfile object, in case that's needed later (e.g. for file metadata). This way this class has only a dependency to the IFileSaveService, and not to the fileManager. Also, it's likely we need to know other things about the text file (e.g. file size or something) in the future, this way, you do not need to create a new manager method for everything.